### PR TITLE
[13.0][ADD] mrp_unbuild_done_message

### PR DESCRIPTION
**Pull request by Anxo82**

### CODE REVIEW

**Comment by Anxo82** [comment]
@dalonsod @ChristianSantamaria  Review please.
Thanks

**Comment by Anxo82** [comment]
@dalonsod @ChristianSantamaria Review please.
Thanks

File: mrp_unbuild_done_message/data/mail_template_data.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?xml version='1.0' encoding='utf-8'?>
+<odoo>
+    <data noupdate="1">
+        <record id="mail_template_unbuild_done_message" model="mail.template">
+            <field name="name">Unbuild done notification</field>
+            <field name="email_from">${(False if not object else object.company_id.email or '') | safe}</field>
+            <field name="email_to">${ctx.get('email_to', '')}</field>
+            <field name="model_id" ref="mrp.model_mrp_unbuild"/>
+            <field name="subject">Unbuild Done</field>
```
Comment (dalonsod) [inline mrp_unbuild_done_message/data/mail_template_data.xml]: Could you add unbuild name to email subject?

**Comment by Anxo82** [comment]
@dalonsod Review please.
Thanks

File: mrp_unbuild_cust_alu_analytics/readme/CONTRIBUTORS.rst [hunk 2]
```diff
@@ -1,3 +1,4 @@
 * Adrian Resua <adrian.resua@solvos.es>
 * David Alonso <david.alonso@solvos.es>
 * Christian Santamaria <christian.santamaria@solvos.es>
+* Ángel Rivas <angel.rivas@solvos.es>
```
Comment (dalonsod) [inline mrp_unbuild_cust_alu_analytics/readme/CONTRIBUTORS.rst:4]: Once you've added yourself to contributors, re-creating `README.rst` is required.

Comment (Anxo82) [reply mrp_unbuild_cust_alu_analytics/readme/CONTRIBUTORS.rst:4]: @dalonsod Done!

File: mrp_unbuild_done_message/i18n/es.po [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+"                        <br/><br/>\n"
+"                        Please review.<br/><br/>\n"
+"                        Thanks.\n"
+"                    </p>\n"
+"                </div>\n"
+"            "
+msgstr ""
+"<div style=\"font-size:13px;font-family:&quot;Lucida Grande&quot;, Helvetica, Verdana, Arial, sans-serif;margin:0px;padding: 0px;\">\n"
+"                    <p style=\"margin:0px;font-size:13px;font-family:&quot;Lucida Grande&quot;, Helvetica, Verdana, Arial, sans-serif;padding: 0px; font-size: 13px;\">\n"
+"                        El usuario ${ctx.get('user_name', '')} ha completado el proceso de deconstrucción para: ${(object.name or '')}.\n"
```
Comment (dalonsod) [inline mrp_unbuild_done_message/i18n/es.po]: Better use "despiece"

File: mrp_unbuild_done_message/i18n/es.po [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+"                        Please review.<br><br>\n"
+"                        Thanks.\n"
+"                    </p>\n"
+"                </div>\n"
+"            "
+
+#. module: mrp_unbuild_done_message
+#: model:mail.template,subject:mrp_unbuild_done_message.mail_template_unbuild_done_message
+msgid "Unbuild ${(object.name or '')} done"
+msgstr "Deconstrucción ${(object.name or '')} realizada"
```
Comment (dalonsod) [inline mrp_unbuild_done_message/i18n/es.po]: Better use "Despiece"

File: mrp_unbuild_cust_alu_analytics/models/mrp_unbuild_process_type.py [hunk 5]
```diff
@@ -19,3 +19,4 @@ class MrpUnbuildProcessType(models.Model):
         inverse_name='process_type_id',
         string='Lists of Materials'
     )
+    send_notification = fields.Boolean(string="Send notification when unbuild is completed", default=False)
```
Comment (dalonsod) [inline mrp_unbuild_cust_alu_analytics/models/mrp_unbuild_process_type.py:22]: `es` translation is missing for this field

File: mrp_unbuild_done_message/models/mrp_unbuild.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+from odoo import models
+
+class MrpUnbuild(models.Model):
+    _inherit = "mrp.unbuild"
+
+    def send_notification_email(self):
+
+        if self.state == 'done':
+            mrp_users = self.env.ref('mrp.group_mrp_manager').users
```
Comment (dalonsod) [inline mrp_unbuild_done_message/models/mrp_unbuild.py]: Checked in AluTest environment, an error is raised if user has no mail, we should exclude them:

imagen

Suggested (not tested):
```suggestion
            mrp_users = self.env.ref('mrp.group_mrp_manager').users.filtered(lambda x: x.email)
```

**Review by dalonsod — CHANGES_REQUESTED** [review]
See comments

**Comment by dalonsod** [comment]
And other thing @Anxo82 , when a mail is sent, its related information is added to unbuild activity, like other models do. In that case, could you provide a screenshot?

**Comment by Anxo82** [comment]
imagen

**Comment by Anxo82** [comment]
@dalonsod Please review
Thanks.

File: mrp_unbuild_done_message/i18n/es.po (hunk 3 above)
Comment (dalonsod) [reply mrp_unbuild_done_message/i18n/es.po]: Still missing this

File: mrp_unbuild_done_message/models/mrp_unbuild.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+            template = self.env.ref('mrp_unbuild_done_message.mail_template_unbuild_done_message')
+
+            template.sudo().with_context(
+                email_to=user_emails,
+                user_name=self.env.user.name
+            ).send_mail(self.id, force_send=True)
+
+            self.message_post(
+                body=_("Notification email sent to: %s") % user_emails
```
Comment (dalonsod) [inline mrp_unbuild_done_message/models/mrp_unbuild.py]: More accurate this:
```suggestion
                body=_("Done unbuild notification email sent to: %s") % user_emails
```

**Review by dalonsod — CHANGES_REQUESTED** [review]
See two new comments

**Comment by Anxo82** [comment]
@dalonsod Please review
Thanks.

**Comment by dalonsod** [comment]
:+1: will test it ASAP!

**Comment by Anxo82** [comment]
@dalonsod Review please.
Thanks.

File: mrp_unbuild_done_message/models/mrp_unbuild.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+                if user_emails:
+                    template = self.env.ref('mrp_unbuild_done_message.mail_template_unbuild_done_message')
+
+                    template.sudo().with_context(
+                        email_to=user_emails,
+                        user_name=self.env.user.name
+                    ).send_mail(self.id, force_send=True)
+
+                    self.message_post(
+                        body=_("Notification email sent to: %s") % user_emails
```
Comment (dalonsod) [inline mrp_unbuild_done_message/models/mrp_unbuild.py]: More specific message that refers to "closed unbuild" is been lost

File: mrp_unbuild_done_message/models/mrp_unbuild.py [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+                    )
+                else:
+                    self.message_post(body=_("No users to notify in the warehouse."))
```
Comment (dalonsod) [inline mrp_unbuild_done_message/models/mrp_unbuild.py]: Please refer to specific process (closed unbuild)

File: mrp_unbuild_done_message/models/stock_warehouse.py [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+
+from odoo import models, fields
+
+class StockWarehouse(models.Model):
+    _inherit = 'stock.warehouse'
+
+    unbuild_notify_users = fields.Many2many('res.users', 
+    string="Users to Notify for Unbuilds", 
+    help="Users to whom to send closure email for parts associated with this warehouse", 
+    domain=[('share', '=', False)]
```
Comment (dalonsod) [inline mrp_unbuild_done_message/models/stock_warehouse.py]: Check indentation, it's seems to be not standard

File: mrp_unbuild_done_message/models/stock_warehouse.py [hunk 11]
```diff
@@ -0,0 +1,13 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import models, fields
+
+class StockWarehouse(models.Model):
+    _inherit = 'stock.warehouse'
+
+    unbuild_notify_users = fields.Many2many('res.users', 
+    string="Users to Notify for Unbuilds",
```
Comment (dalonsod) [inline mrp_unbuild_done_message/models/stock_warehouse.py]: Please add context:
```suggestion
    unbuild_close_notify_users = fields.Many2many('res.users', 
    string="Users to Notify for Unbuilds closure", 
```

**Review by dalonsod — CHANGES_REQUESTED** [review]
See comments

**Comment by Anxo82** [comment]
@dalonsod Review please.
Thanks.

**Comment by dalonsod** [comment]
OK, I'll test these new improvements ASAP

**Comment by dalonsod** [comment]
New Many2many field is badly placed:
after:
imagen

before:
imagen

Placed after "Dirección" (partner_id):
imagen

Make the suggested change

**Comment by dalonsod** [comment]
Review done in AluTest environment :+1: 

Once required change is done, please make pending squash.

**Comment by Anxo82** [comment]
@dalonsod Review please.
Thanks

**Comment by dalonsod** [comment]
One last change... please add to `mrp_unbuild_done_message` commit description (not title) the description provided for this addon.

**Comment by Anxo82** [comment]
@dalonsod Review please.
Thanks

**Review by dalonsod — APPROVED** [review]
:+1:

**Comment by dalonsod** [comment]
@ChristianSantamaria merge made at `slv-manufacture` FYI

**Pull request merged**